### DISCUSSION TITLE
Function like FermionSpinSum, but for unsquared amplitudes?

### DISCUSSION THREAD

**Discussion by oscatwilson** [FeynCalc/feyncalc#210] [Q&A]
Hi,

I'm trying to use FeynCalc to model Dirac fermions with simple interactions terms. I'm using FeynRules to create the model fermions and Lagrangian, then using FeynArts and FeynCalc to calculate the amplitudes. 

This fermion model uses a four-fermion interaction term (φ γ^μ ψ)^2 (where φ is the adjoint spinor of ψ and γ are the Dirac matrices. This is written in FeynRules as `psibar.Ga[nu].psi psibar.Ga[nu].psi`

I can sum over these and calculate the squared amplitude of the interactions using FermionSpinSum, which works fine. But in this project I specifically need to calculate and analyse the un-squared amplitude - the final sign of the amplitude is important for analysis later.

I have tried using DiracSimplify, ExpandScalarProduct, Calc[SP[]], Simplify, DiracTrace and SpinorChainTrick but none of these will evaluate the fermion chains in the amplitude. I have looked through the FeynCalc guide and various example scattering models too, and haven't been able to find anything that would work.

Is there any function in FeynCalc, or even FeynArts, that is able to exaluate these fermion chains, but for an un-squared amplitude rather than the squared amplitude? Or is there a problem in my FeynRules model of the interaction which is preventing the amplitude from being summed? Any light you can shed on this would be very helpful.

I have attached the files I'm using in case they are useful. I'm currently using FeynCalc 9.3.1, FeynArts 3.11, and Mathematica 13.1. Any light you can shed on this would be very helpful.

[Simple Interacting Fermion Model.zip](https://github.com/FeynCalc/feyncalc/files/10578191/Simple.Interacting.Fermion.Model.zip)

**Comment by vsht** [comment]
Hi,

> Is there any function in FeynCalc, or even FeynArts, that is able to exaluate these fermion chains, but for an un-squared amplitude rather than the squared amplitude? Or is there a problem in my FeynRules model of the interaction which is preventing the amplitude from being summed? Any light you can shed on this would be very helpful.

do you mean writing down the amplitude for a particular helicity configuration? Or picking an explicit representation of the Dirac spinors
and reducing the matrix element to a c-number?

There is no dedicated function for that, but you could use `DiracReduce` and `FCDiracIsolate` to extract all possible Dirac structures appearing 
in the amplitude and then use hand-written replacement rules for the evaluation of the Dirac chains. At least this is how I would do it in such a situation.

Cheers,
Vladyslav
